modules/apps/foundation/address-formatter/address-formatter-api/src/main/java/com/liferay/address/formatter/util/AddressFormatterUtil.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.address.formatter.util;
+
+import com.liferay.address.formatter.AddressFormatter;
+import com.liferay.osgi.service.tracker.collections.map.ServiceTrackerMap;
+import com.liferay.osgi.service.tracker.collections.map.ServiceTrackerMapFactory;
+import com.liferay.portal.kernel.model.Address;
+import com.liferay.portal.kernel.model.Country;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+
+/**
+ * @author Pei-Jung Lan
+ */
+@Component(immediate = true, service = AddressFormatterUtil.class)
+public class AddressFormatterUtil {
+
+	public static String format(Address address) {
+		Country country = address.getCountry();
+
+		String a2 = country.getA2();
+
+		AddressFormatter addressFormatter = getAddressFormatter(a2);
+
+		return addressFormatter.format(address);
+	}
+
+	public static AddressFormatter getAddressFormatter(String countryA2) {
```

Comment:
I see. Where I saw a limitation is in deciding the format just based on the country. But I can see how it could work by delegating the Formatter the decision of choosing between different representations for the same country. That can work as long as we don't want to, for example, represent an address from Japan like with the usual US format (which I think should be uncommon).

Another point is that in the ticket we also wrote that the formatter should also have access to the site's locale and the locale of the user viewing the address. I don't remember why we wrote this and it might be worth double checking with our colleagues in Japan and China. We could just make them available in the API, however passing these might mean that we are expecting the Formatters to overlap with each other in responsabilities, don't you think?